### DISCUSSION TITLE
Does the new flat config always deep merge any conflicting object?

### DISCUSSION THREAD

**Discussion by xsjcTony** [eslint/eslint#17689] [Q&A] [answered]
In the document [Cascading configuration objects](https://eslint.org/docs/latest/use/configure/configuration-files-new#cascading-configuration-objects), I can see that `languageOptions.globals` will be merged in case of conflict.

I want to know:
1. Is any conflicting object going to be merged? since `languageOptions.globals` is a field provided by the official configuration, but what if it's a custom object in `settings`?
2. Will those conflicting objects in `settings` be deeply merged, or just a shallow merge (e.g. by `...`)?

Example with `eslint-plugin-import`
```js
export default [
  {
    files: ['**/*.{ts,tsx}],
    settings: {
      'import/parsers': {
        '@typescript-eslint/parser': ['.ts', '.tsx']
      }
    }
  },
  {
    files: ['**/*.{ts,tsx,vue}],
    settings: {
      'import/parsers': {
        'vue-eslint-parser': ['.vue']
      }
    }
  }
]
```

Will the `'@typescript-eslint/parser': ['.ts', '.tsx']` also apply to `*.vue` files? or do I have to define it again?
What about even deeper nested objects?

**Comment by Rec0iL99** [comment, marked as answer]
Hi @xsjcTony, you can use the [`--print-config`](https://eslint.org/docs/latest/use/command-line-interface#--print-config) CLI option to view the final configuration that will be used for a file. In this case, I believe the `settings` key would be the following:

```js
"settings": {
    "import/parsers": {
      "@typescript-eslint/parser": [
        ".ts",
        ".tsx"
      ],
      "vue-eslint-parser": [
        ".vue"
      ]
    }
  },
```
